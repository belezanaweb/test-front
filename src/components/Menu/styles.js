import styled from 'styled-components'

export const Wrapper = styled.div`
  border: 1px solid red;
  background-color: #fff;
`

export const Menu = styled.ul`
  border: 1px solid red;
`

export const Option = styled.li`
  border: 1px solid red;
`
