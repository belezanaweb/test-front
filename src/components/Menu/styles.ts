import styled, { css, DefaultTheme } from 'styled-components'
import media from 'styled-media-query'

export const Nav = styled.nav`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 0.8fr 1fr 1fr;
    box-shadow: 1px 1px 5px rgba(0, 0, 29, 0.22);
    background: ${theme.colors.white};
    ${media.greaterThan('medium')`
      display: flex;
      flex-direction: column;
      border: 0;
      grid-gap: 0.3rem;
      padding: 0.5rem;
      a:not(:last-child) {
        border-bottom: 0.1rem solid ${theme.colors.lightGray};
      }
    `}
  `}
`

const linkModifiers = {
  default: (theme: DefaultTheme) => css`
    color: ${theme.colors.mediumGray};
  `,
  active: (theme: DefaultTheme) => css`
    color: ${theme.colors.primary};
  `
}

type LinkProps = {
  isActive?: boolean
}

export const Link = styled.a<LinkProps>`
  ${({ theme, isActive }) => css`
    cursor: pointer;
    text-decoration: none;
    display: flex;
    align-items: center;
    padding: ${theme.spacings.small} 0;
    font-family: ${theme.font.family};
    font-size: ${theme.font.sizes.xsmall};
    font-weight: ${theme.font.bold};
    transition: color, ${theme.transition.default};
    &:hover {
      color: ${theme.colors.primary};
    }

    ${media.lessThan('medium')`
      justify-content: center;
      flex: 1;
    `}

    ${media.greaterThan('medium')`
      padding: ${theme.spacings.small} 1rem;
    `}

    ${!isActive && linkModifiers.default(theme)};
    ${isActive && linkModifiers.active(theme)}
  `}
`
