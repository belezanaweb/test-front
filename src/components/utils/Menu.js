/** @jsx jsx */
import { jsx } from '@emotion/core'
import { Link } from 'react-router-dom'
import styled from '@emotion/styled'

const MenuBody = styled('nav')`
  margin: 32px 13px 0 14px;
  height: 72px;
  width: 387px;
  border-radius: 3px;
  background-color: #FFF;
  box-shadow: 1px 1px 5px 0 rgba(0,0,29,0.22);

  ul {
    display: flex;
    justify-content: space-around;

    li {
      margin: 44px 29px 12px 0;
      list-style: none;

      .sacola {
        height: 16px;
        width: 61px;
        color: #FF7800;
        font-family: "Helvetica Neue";
        font-size: 13px;
        font-weight: 700;
        letter-spacing: NaNpx;
        line-height: 16px;
        text-align: center;
      }
    }
  }

`
const Menu = () => {
  return (
    <MenuBody>
      <ul>
        <li>
          <Link className='sacola'>Sacola</Link>
        </li>
        <li>
          <Link>Pagamento</Link>
        </li>
        <li>
          <Link>Confirmação</Link>
        </li>
      </ul>
    </MenuBody>
  )
}

export default Menu