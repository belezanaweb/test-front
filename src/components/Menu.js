import React from 'react';
import { withRouter } from "react-router";

import { TabsCSS } from '../styles';

function Menu(props) {

  const { Container, Tabs, Tab } = TabsCSS;
  const path = props.location.pathname.replace("/","");
  const tabs = [
    { id: "cart", text: "Sacola" },
    { id: "payment", text: "Pagamento" },
    { id: "success", text: "Confirmação" }
  ]

  return(
    <Container>
      <Tabs>
        {
          tabs.map(
            tab =>
              <Tab
                key={tab.id}
                active={ path === tab.id ? true : false }
              >
                {tab.text}
              </Tab>
          )
        }
      </Tabs>
    </Container>
  )
}

export default withRouter(Menu);
