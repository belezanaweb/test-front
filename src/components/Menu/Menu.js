import React, { Children } from 'react';

import { StyledHeader, StyledNav, StyledLogo, StyledList } from './styles';

const Menu = ({ children }) => {
  return (
    <StyledHeader>
      <StyledNav>
        <StyledLogo>Beleza na Web</StyledLogo>
        <StyledList>
          {Children.map(children, (child) => (
            <li key={child.key}>{child}</li>
          ))}
        </StyledList>
      </StyledNav>
    </StyledHeader>
  );
};

export default Menu;
