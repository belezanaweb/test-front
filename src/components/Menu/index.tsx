import {
  CheckSquare,
  CreditCardFront,
  ShoppingBag
} from '@styled-icons/boxicons-regular'
import React, { useState } from 'react'
import { Redirect } from 'react-router-dom'
import * as S from './styles'

export type ProfileMenuProps = {
  activeLink?: '/' | '/payment' | '/success' | string
}

const Menu = ({ activeLink }: ProfileMenuProps) => {
  const [goToBag, setGoToBag] = useState(false)
  const [goToPayment, setGoToPayment] = useState(false)
  const [sucess, setSucess] = useState(false)
  return (
    <S.Nav>
      <S.Button
        onClick={() => {
          setGoToPayment(false)
          setSucess(false)
          setGoToBag(true)
        }}
        isActive={activeLink === '/'}
        title="Sacola"
      >
        <S.Icon>
          <ShoppingBag size={24} />
        </S.Icon>
        <span>SACOLA</span>
      </S.Button>

      <S.Button
        onClick={() => {
          setSucess(false)
          setGoToBag(false)
          setGoToPayment(true)
        }}
        isActive={activeLink === '/payment'}
        title="Pagamento"
      >
        <S.Icon>
          <CreditCardFront size={24} />
        </S.Icon>
        <span>PAGAMENTO</span>
      </S.Button>

      <S.Button
        onClick={() => {
          setGoToBag(false)
          setGoToPayment(false)
          setSucess(true)
        }}
        isActive={activeLink === '/success'}
        title="Confirmação"
      >
        <S.Icon>
          <CheckSquare size={24} />
        </S.Icon>
        <span>CONFIRMAÇÃO</span>
      </S.Button>
      {goToBag && <Redirect to="/" />}
      {goToPayment && <Redirect to="/payment" />}
      {sucess && <Redirect to="/success" />}
    </S.Nav>
  )
}

export default Menu
