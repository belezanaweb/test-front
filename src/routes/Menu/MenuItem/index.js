import React from 'react'

import { Container, Name } from './styles'
import { Link } from 'react-router-dom'

function MenuItem({ menu }) {
  return (
    <Container>
      <Name selected={true}>
        <Link to={menu.path}>{menu.name}</Link>
      </Name>
    </Container>
  )
}

export default MenuItem
