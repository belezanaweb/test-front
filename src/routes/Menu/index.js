import React from 'react'

import { Container } from './styles'
import MenuItem from './MenuItem'

function Menu({ menus }) {
  return (
    <Container>
      {menus.length > 0 &&
        menus.map((menu) => {
          console.log('teste', menu)
          return <MenuItem menu={menu} />
        })}
    </Container>
  )
}

export default Menu
