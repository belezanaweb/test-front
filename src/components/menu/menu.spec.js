import React from 'react';
import Menu from '.';

const links = [
  {
    title: 'Sacola',
    path: '/'
  },
  {
    title: 'Pagamento',
    path: '/pagamento'
  },
  {
    title: 'Confirmação',
    path: '/confirmacao'
  }
];

describe('Menu', () => {
  it('should match a snapshot', () => {
    const wrapper = getStyledWrapperJSON(Menu, {
      children: links.map(({ title, path }, index) => (
        <a key={path} href={path} className={index === 0 && 'active'}>
          {title}
        </a>
      ))
    });

    expect(wrapper).toMatchSnapshot();
  });
});
