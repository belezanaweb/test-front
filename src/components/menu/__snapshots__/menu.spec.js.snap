// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match a snapshot 1`] = `
.c0 {
  background-color: #fff;
  box-shadow: 1px 1px 5px 0 rgba(0,0,29,0.22);
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  width: 100%;
}

.c3 {
  margin: 0 auto;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  list-style: none;
  max-width: 375px;
}

.c3 a {
  display: block;
  padding: 0.8rem 0.7rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-weight: 700;
  font-size: 0.844rem;
  color: #ccc;
  -webkit-transition: color 0.2s ease;
  transition: color 0.2s ease;
}

.c3 a.active {
  color: #ff6c00;
}

.c2 {
  display: none;
}

@media (min-width:768px) {
  .c0 {
    padding: 1rem 0;
  }
}

@media (min-width:768px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    max-width: 998px;
    margin: 0 auto;
    padding: 0 1rem;
  }
}

@media (min-width:768px) {
  .c3 {
    margin: 0 -1.2rem 0 0;
  }
}

@media (min-width:375px) {
  .c3 a {
    padding: 0.8rem 1.2rem;
  }
}

@media (min-width:768px) {
  .c2 {
    width: 80px;
    height: 40px;
    margin: 0;
    display: block;
    background: url(logo.svg) no-repeat 0px -20px;
    background-size: cover;
    text-indent: -9999px;
    font-size: 0;
  }
}

<header
  className="c0"
>
  <nav
    className="c1"
  >
    <h1
      className="c2"
    >
      beleza na web
    </h1>
    <ul
      className="c3"
    >
      <li>
        <a
          className="active"
          href="/"
        >
          Sacola
        </a>
      </li>
      <li>
        <a
          className={false}
          href="/pagamento"
        >
          Pagamento
        </a>
      </li>
      <li>
        <a
          className={false}
          href="/confirmacao"
        >
          Confirmação
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
